dojo/unittests/tools/test_securecodebox_parser.py

```diff
@@ -0,0 +1,82 @@
+from django.test import TestCase
+from dojo.tools.securecodebox.parser import SecureCodeBoxParser
+from dojo.models import Engagement, Product, Test
+
+
+class TestSecureCodeBoxParser(TestCase):
+
+    def get_test(self):
+        test = Test()
+        test.engagement = Engagement()
+        test.engagement.product = Product()
+        return test
+
+    def test_scb_parser_without_file_has_no_findings(self):
+        parser = SecureCodeBoxParser()
+        findings = parser.get_findings(None, self.get_test())
+        self.assertEqual(0, len(findings))
+
+    def test_scb_parser_with_no_vuln_has_no_findings(self):
+        testfile = open("dojo/unittests/scans/securecodebox/scb_zero_vul.json")
+        parser = SecureCodeBoxParser()
+        findings = parser.get_findings(testfile, self.get_test())
+        testfile.close()
+        self.assertEqual(0, len(findings))
+
+    def test_scb_parser_with_one_criticle_vuln_has_one_findings(self):
+        testfile = open("dojo/unittests/scans/securecodebox/scb_one_vul.json")
+        parser = SecureCodeBoxParser()
+        findings = parser.get_findings(testfile, self.get_test())
+        testfile.close()
+        self.assertEqual(1, len(findings))
+        self.assertEqual("High", findings[0].severity)
+
+    def test_scb_parser_with_many_vuln_has_many_findings(self):
+        testfile = open("dojo/unittests/scans/securecodebox/scb_many_vul.json")
+        parser = SecureCodeBoxParser()
+        findings = parser.get_findings(testfile, self.get_test())
+        testfile.close()
+        self.assertEqual(5, len(findings))
+        self.assertEqual(findings[0].severity, "Info")
+        self.assertEqual(findings[0].title, "ssh")
+        self.assertEqual(findings[0].description,
+                         "Port 22 is open using tcp protocol.")
+        self.assertEqual(
+            findings[0].unsaved_endpoints[0].host, "scanme.nmap.org")
+        self.assertEqual(findings[0].unsaved_endpoints[0].protocol, "tcp")
+        self.assertEqual(findings[0].unsaved_endpoints[0].path, "")
```

Comment:
Maybe not the best option here. Better to have `path` at `None` as this attribute is null-able.

Reply:
path, query and fragment are no longer some default values but None if they are empty